File: app/src/main/java/com/denchic45/kts/data/model/room/GroupWithCuratorAndSpecialtyEntity.java
package com.denchic45.kts.data.model.room;

import androidx.room.Embedded;
import androidx.room.Ignore;
import androidx.room.Relation;

import com.denchic45.kts.data.model.EntityModel;

public class GroupWithCuratorAndSpecialtyEntity implements EntityModel {

    @Embedded
    private GroupEntity groupEntity;

    @Relation(
            parentColumn = "uuid_curator",
            entityColumn = "uuid_user"
    )
    private UserEntity curatorEntity;

    @Relation(
            parentColumn = "specialtyUuid",
            entityColumn = "specialtyUuid"
    )
    private SpecialtyEntity specialtyEntity;

    public GroupWithCuratorAndSpecialtyEntity() {

    }

    @Ignore
    public GroupWithCuratorAndSpecialtyEntity(GroupEntity groupEntity, UserEntity curatorEntity, SpecialtyEntity specialtyEntity) {
        this.groupEntity = groupEntity;
        this.curatorEntity = curatorEntity;
        this.specialtyEntity = specialtyEntity;
    }

    public SpecialtyEntity getSpecialtyEntity() {
        return specialtyEntity;
    }

    public void setSpecialtyEntity(SpecialtyEntity specialtyEntity) {
        this.specialtyEntity = specialtyEntity;
    }

    public GroupEntity getGroupEntity() {
        return groupEntity;
    }

    public void setGroupEntity(GroupEntity groupEntity) {
        this.groupEntity = groupEntity;
    }

    public UserEntity getCuratorEntity() {
        return curatorEntity;
    }

    public void setCuratorEntity(UserEntity curatorEntity) {
        this.curatorEntity = curatorEntity;
    }
}

File: app/src/main/java/com/denchic45/kts/data/model/room/GroupWithCuratorAndSpecialtyEntity.kt
package com.denchic45.kts.data.model.room

import androidx.room.Embedded
import androidx.room.Relation
import com.denchic45.kts.data.domain.model.EntityModel

data class GroupWithCuratorAndSpecialtyEntity(
    @Embedded
    val groupEntity: GroupEntity,
    @Relation(parentColumn = "curator_id", entityColumn = "user_id")
    val curatorEntity: UserEntity,
    @Relation(parentColumn = "specialty_id", entityColumn = "specialty_id")
    val specialtyEntity: SpecialtyEntity
) : EntityModel

data class GroupWithCuratorAndStudentsEntity(
    @Embedded
    val groupEntity: GroupEntity,
    @Relation(parentColumn = "curator_id", entityColumn = "user_id")
    val curatorEntity: UserEntity,
    @Relation(parentColumn = "group_id", entityColumn = "user_group_id")
    val studentEntities: List<UserEntity>
) : EntityModel